operator/controllers/metrics/keptnmetric_controller.go

```diff
@@ -61,12 +73,61 @@ func (r *KeptnMetricReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	return ctrl.Result{}, nil
+	if time.Now().Before(metric.Status.LastUpdated.Add(time.Second * time.Duration(metric.Spec.FetchIntervalSeconds))) {
+		r.Log.Info("Metric has not been updated for the configured interval. Skipping")
+		return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	evaluationProvider, err := r.fetchProvider(ctx, types.NamespacedName{Name: metric.Spec.Provider.Name, Namespace: metric.Namespace})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info(err.Error() + ", ignoring error since object must be deleted")
+			return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+		r.Log.Error(err, "Failed to retrieve the provider")
+		return ctrl.Result{}, nil
+	}
+	// load the provider
+	provider, err2 := providers.NewProvider(metric.Spec.Provider.Name, r.Log, r.Client)
+	if err2 != nil {
+		//r.recordEvent("Error", metric, "ProviderNotFound", "provider was not found")
+		r.Log.Error(err2, "Failed to get the correct Metric Provider")
+		return ctrl.Result{Requeue: false}, err2
+	}
+
+	objective := klcv1alpha2.Objective{
+		Name:  metric.Name,
+		Query: metric.Spec.Query,
+	}
+	value, rawValue, err := provider.EvaluateQuery(ctx, objective, *evaluationProvider)
+	if err != nil {
+		//r.recordEvent("Error", metric, "EvaluationFailed", "evaluation failed")
+		r.Log.Error(err, "Failed to evaluate the query")
+		return ctrl.Result{Requeue: false}, err
+	}
+	metric.Status.Value = value
+	metric.Status.RawValue = rawValue
```

Comment:
given the max size of a CRD is 1.5MB ([etcd docs](https://etcd.io/docs/v3.5/dev-guide/limit/#request-size-limit)), I suggest to hard cut the rawValue size to 1MB